host/class/uvc/usb_host_uvc_2/uvc_descriptor_printing.c

```diff
@@ -0,0 +1,387 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "usb/usb_host.h"
+#include "usb/uvc_host.h"
+#include "uvc_types_priv.h"
+
+#define TERMINAL_INPUT_CAMERA_TYPE      0x0201
+#define TERMINAL_INPUT_COMPOSITE_TYPE   0x0401
+#define ITT_MEDIA_TRANSPORT_INPUT       0x0202
+
+static enum uvc_int_subclass_code interface_sub_class = UVC_SC_VIDEOCONTROL;
+
+static void unknown_desc(const usb_standard_desc_t *header)
+{
+    printf(" *** Unknown Descriptor Length: %d Type: %d ***\n",
+           header->bLength, header->bDescriptorType);
+}
+
+static void print_cs_endpoint_desc(const usb_standard_desc_t *_desc)
+{
+    uvc_vc_interrupt_ep_desc_t *class_desc = (uvc_vc_interrupt_ep_desc_t *)_desc;
+    printf("\t\t*** Class-specific Interrupt Endpoint Descriptor ***\n");
+    printf("\t\tbLength 0x%x\n", class_desc->bLength);
+    printf("\t\tbDescriptorType 0x%x\n", class_desc->bDescriptorType);
+    printf("\t\tbDescriptorSubType %d\n", class_desc->bDescriptorSubType);
+    printf("\t\twMaxTransferSize %d\n", class_desc->wMaxTransferSize);
+}
+
+static void print_class_header_desc(const usb_standard_desc_t *_desc)
+{
+    if (interface_sub_class == UVC_SC_VIDEOCONTROL) {
+        const uvc_vc_header_desc_t *desc = (const uvc_vc_header_desc_t *) _desc;
+        printf("\t*** Class-specific VC Interface Descriptor ***\n");
+        printf("\tbLength 0x%x\n", desc->bLength);
+        printf("\tbDescriptorType 0x%x\n", desc->bDescriptorType);
+        printf("\tbDescriptorSubType %u\n", desc->bDescriptorSubType);
```

Comment:
Here are 2 similar fields, one is printed in hex, the other one in decimal. Similarly, for a few following fields of those types.